sm/sm.h

```diff
@@ -23,6 +23,7 @@
 #define SBI_SM_NOT_IMPLEMENTED  1111
 
 /* error codes */
+#define ENCLAVE_NO_FREE_RESOURCE            (enclave_ret_t)-3U
```

Comment:
How are we assigning error values? They are all U right now, so the scheme we have doesn't make much sense.
Can we just switch this to all positive integer values?

Reply:
My intention was to indicate that two negative ones are temporary for transition, which means that they should go away when all required error codes are identified and defined. All valid error codes should be positive.

Reply:
Makes sense, we'll keep unusual error codes for special cases (not implemented, unknown error condition).
They aren't safe to consider as negative however, since most things handling them are treating them as unsigned.